src/sidebar/components/share-dialog.js

```diff
@@ -8,15 +8,15 @@ function ShareDialogController($scope, $element, analytics, store) {
 
   function updateViaLink(frames) {
     if (!frames.length) {
-      self.viaPageLink = '';
+      self.sharePageLink = '';
       return;
     }
 
     // Check to see if we are on a via page. If so, we just return the URI.
     if (frames[0].uri.indexOf(VIA_PREFIX) === 0) {
-      self.viaPageLink = frames[0].uri;
+      self.sharePageLink = frames[0].uri;
     } else {
-      self.viaPageLink = VIA_PREFIX + frames[0].uri;
+      self.sharePageLink = 'https://hyp.is/go?url='+frames[0].uri;
```

Comment:
Hardcoding the URL is a reasonable first step. I explained in a separate comment how I think we should get the "hyp.is" link from the service.

Since the URL is appearing in a query string param here, it ought to be URL escaped. One way to do this would be to use the existing functions for working with URL templates that get used for handling URL templates returned by `/api/links`:

```diff
diff --git a/src/sidebar/components/share-dialog.js b/src/sidebar/components/share-dialog.js
index 76f1f45e..fa2df4ac 100644
--- a/src/sidebar/components/share-dialog.js
+++ b/src/sidebar/components/share-dialog.js
@@ -1,5 +1,8 @@
 'use strict';
 
+var { replaceURLParams } = require('../util/url-util');
+
+var SHARE_URL_TEMPLATE = 'https://hyp.is/go?url=:url';
 var VIA_PREFIX = 'https://via.hypothes.is/';
 
 // @ngInject
@@ -16,7 +19,8 @@ function ShareDialogController($scope, $element, analytics, store) {
     if (frames[0].uri.indexOf(VIA_PREFIX) === 0) {
       self.sharePageLink = frames[0].uri;
     } else {
-      self.sharePageLink = 'https://hyp.is/go?url='+frames[0].uri;
+      var { url } = replaceURLParams(SHARE_URL_TEMPLATE, { url: frames[0].uri });
+      self.sharePageLink = url;
     }
   }
```

Reply:
I get your point. However it turns out that we don't actually need this branch because when on a Via page, the client believes the page's URL is the original URL, not `https://<proxy service>/<original URL>` because of the `<link rel="canonical">` tag in the page.